front/components/assistant/AgentEditBar.tsx

```diff
@@ -0,0 +1,167 @@
+  tags: TagType[];
+  mutateAgentConfigurations: () => Promise<any>;
+};
+
+export const AgentEditBar = ({
+  onClose,
+  selectedAgents,
+  owner,
+  tags,
+  mutateAgentConfigurations,
+}: AgentEditBarProps) => {
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+  const [tagSearch, setTagSearch] = useState<string>("");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const updateAgentTags = useUpdateAgentTags({
+    owner,
+  });
+
+  const filteredTags = tags
+    .filter((t) => isBuilder(owner) || t.kind !== "protected")
+    .filter((a) => {
+      return subFilter(tagSearch, a.name.toLowerCase());
+    })
+    .sort((a, b) => {
+      if (tagSearch) {
+        return compareForFuzzySort(
+          tagSearch,
+          a.name.toLowerCase(),
+          b.name.toLowerCase()
+        );
+      } else {
+        return tagsSorter(a, b);
+      }
+    });
+
+  return (
+    <>
+      <DeleteAssistantsDialog
+        owner={owner}
+        agentConfigurations={selectedAgents}
+        isOpen={isDeleteDialogOpen}
+        onClose={() => setIsDeleteDialogOpen(false)}
+        onSave={() => {
+          onClose();
+          setIsDeleteDialogOpen(false);
+        }}
+      />
+
+      <div className="border-1 mb-2 flex flex-row items-center gap-2 rounded-xl bg-muted-background p-2 dark:bg-muted-background-night">
+        <Button
+          size="xs"
+          variant="outline"
+          disabled={isLoading}
+          label="Close edition"
+          icon={XMarkIcon}
+          onClick={onClose}
+        />
+        {isLoading && <Spinner size="xs" variant="dark" />}
+        <div className="flex-1" />
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button
+              size="xs"
+              variant="outline"
+              isSelect
+              icon={TagIcon}
+              label="Tag selection"
+              disabled={selectedAgents.length === 0 || isLoading}
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent
+            dropdownHeaders={
+              <>
+                <DropdownMenuSearchbar
+                  name="tagSearch"
+                  placeholder="Search tags"
+                  value={tagSearch}
+                  onChange={setTagSearch}
+                />
+                <DropdownMenuSeparator />
+              </>
+            }
+          >
+            {filteredTags.map((t) => {
+              return (
+                <DropdownMenuItem
+                  key={t.sId}
+                  onClick={async () => {
+                    setIsLoading(true);
+                    if (
+                      selectedAgents.every((a) =>
+                        a.tags.find((agentTag) => agentTag.sId === t.sId)
+                      )
+                    ) {
+                      for (const a of selectedAgents) {
+                        // Remove
+                        await updateAgentTags(a.sId, {
```

Comment:
Should we have a updateAgentsTags that takes a list of agents and their ids of tags to add/remove?